efs-lib/src/cmr/efs/connection.clj

```diff
@@ -26,19 +49,27 @@
 (defn save-concept
   "Saves a concept to EFS"
   [provider concept-type concept]
-  (let [concept-path (format "%s/%s/%s/%s.r%d.zip" (efs-config/efs-directory) (:provider-id provider) (name concept-type) (:concept_id concept) (:revision_id concept))]
+  (let [concept-path (format "%s/%s/%s/%s/%s.r%d.zip" (efs-config/efs-directory) (:provider-id provider) (name concept-type) (:concept_id concept) (:concept_id concept) (:revision_id concept))]
     (info "Saving concept to EFS at path " concept-path)
     (io/make-parents (io/file concept-path))
     (Files/write (Paths/get concept-path (into-array String [])) (:metadata concept) (into-array OpenOption []))))
 
 (defn get-concept
   "Gets a concept from EFS"
-  ([provider concept-type concept-id])
-  ([provider concept-type concept-id revision-id]))
+  ([provider concept-type concept-id]
+   (get-concept provider concept-type concept-id (get-latest-revision (:provider-id provider) (name concept-type) concept-id)))
+  ([provider concept-type concept-id revision-id]
+   (if (and revision-id
+            (concept-revision-exists (:provider-id provider) (name concept-type) concept-id revision-id))
```

Comment:
1) Any time you have and if and you want to return a nil, you should use (when instead of (if
2) While the probability is low, you are creating a race condition when checking to see if a file exists and then later you try and read it.  The file could get deleted in the time between the successful exists and the time the file gets read.  You should just read the file and create logic to handle the case where it doesn't exist.